filters/ema.hpp

```diff
@@ -38,71 +26,67 @@ class EMAFilter : public Consumer<In>, public Producer<double> {
     }
   }
 
-  ~EMAFilter() {}
-
-  virtual Try<Nothing> consume(const In& in) = 0;
-
-  void setAlpha(double _alpha) {
+  void setAlpha(double_t _alpha) {
     alpha = _alpha;
   }
 
-  double getAlpha() {
+  double_t getAlpha() {
     return alpha;
   }
 
- protected:
+  /*
+  * Calculate EMA and save needed values for next calculation.
+  */
+  double_t calculateEMA(double_t sample, double_t sampleTimestamp) {
+    if (uninitialized) {
+      prevEma = sample;
+      uninitialized = false;
+    }
+    prevEma = exponentialMovingAverageRegular(sample);
```

Comment:
If the EMA should be counted by regular or irregular means should be resolved by strategy/dependency injection.